File: ios_app/Common/Extensions/ColorEnum.swift
import SwiftUI
import shared

extension ColorEnum {
    
    func toColor() -> Color {
        switch self {
        case .white: .white
        case .black: .black
            
        case .red: .red
        case .orange: .orange
        case .yellow: .yellow
        case .green: .green
        case .mint: .mint
        case .teal: .teal
        case .cyan: .cyan
        case .blue: .blue
        case .indigo: .indigo
        case .purple: .purple
        case .pink: .pink
        case .brown: .brown
            
        case .gray: .gray
            
        default: fatalError("ColorEnum.toColor() no \(self)")
        }
    }
}

File: android_app/src/main/java/me/timeto/app/ColorEnum.kt
package me.timeto.app

import androidx.compose.ui.graphics.Color
import me.timeto.app.ui.c
import me.timeto.shared.ColorEnum

fun ColorEnum.toColor(): Color = when (this) {

    ColorEnum.white -> c.white
    ColorEnum.black -> c.black

    ColorEnum.red -> c.red
    ColorEnum.orange -> c.orange
    ColorEnum.yellow -> c.yellow
    ColorEnum.green -> c.green
    ColorEnum.mint -> c.mint
    ColorEnum.teal -> c.teal
    ColorEnum.cyan -> c.cyan
    ColorEnum.blue -> c.blue
    ColorEnum.indigo -> c.indigo
    ColorEnum.purple -> c.purple
    ColorEnum.pink -> c.pink
    ColorEnum.brown -> c.brown

    ColorEnum.gray -> c.gray

    ColorEnum.text -> c.teal
    ColorEnum.secondaryText -> c.secondaryText
}
